PatientRecords/DataAccessLayer/GenericRepository.cs

```diff
@@ -0,0 +1,53 @@
+﻿using DataAccessLayer.DTO;
+using DataAccessLayer.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer
+{
+    public class GenericRepository<TModel> : IRepository<TModel, int>
+        where TModel : class
+    {
+        private readonly PatientRecordsContext _context;
+
+        public GenericRepository(PatientRecordsContext context)
+        {
+            _context = context;
+        }
+
+        public async Task CreateAsync(TModel item)
+        {
+            await _context.Set<TModel>().AddAsync(item);
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task DeleteAsync(TModel item)
+        {
+            _context.Set<TModel>().Remove(item);
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task SaveAsync() => await _context.SaveChangesAsync();
```

Comment:
I think this method redundant and should be removed.
@Solinx97 please, remove this method in your repository too.